### PR TITLE
Profiler recipe

### PR DESCRIPTION
**Pull request by ilia-cher**
Summary:
Adding a recipe for profiler

Test Plan:
make html-noplot

### CODE REVIEW

File: recipes_source/recipes/profiler.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+# _convolution               7.171ms      3         [[5, 512, 7, 7], [512, 512, 3, 3], []]
+# max_pool2d_with_indices    7.170ms      1         [[5, 64, 112, 112]]
+# -------------------------  -----------  --------  --------------------------------------
+
+
+######################################################################
+# 4. Use profiler to analyze memory consumption
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# PyTorch profiler can also show the amount memory allocated (freed) for the tensors
```
Comment (ngimel) [inline recipes_source/recipes/profiler.py:144]: What does allocated (freed) mean? Also, a brief explanation of what CPU Mem and Self CPU Mem is would be useful (I take it, similar to time, it's memory allocated by function and its children, or function itself).

File: recipes_source/recipes/profiler.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        model(inputs)
+
+prof.export_chrome_trace("trace.json")
+
+######################################################################
+# User can examine the sequence of profiled operators after loading the trace file
+# in Chrome (``chrome://tracing``):
+#
+# .. image:: ../../_static/img/trace_img.png
+#    :scale: 25 %
```
Comment (jlin27) [inline recipes_source/recipes/profiler.py:203]: Are there additional resources for users to learn more or follow-up topics (e.g. docs, tutorials) we can add as resources?

File: recipes_source/recipes/profiler.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
+"""
+PyTorch Profiler
+====================================
+This recipe explains how to use PyTorch profiler and measure time and
```
[the remark this answers is not in the document]
Comment (jlin27) [reply recipes_source/recipes/profiler.py:4]: nit: add the - measure "the" time and memory consumption...

File: recipes_source/recipes/profiler.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+#
+# Let's see how we can use profiler to analyze execution time
+
+with profiler.profile(record_shapes=True) as prof:
+    with profiler.record_function("model_inference"):
+        model(inputs)
+
+######################################################################
+# Note that we can use ``record_function`` context manager to label
+# arbitrary code ranges with user provided names.
```
Comment (jlin27) [inline recipes_source/recipes/profiler.py:80]: Clarification: Sorry if this is an obvious question, in this example is "model_inference" the label for the code range?

Comment (ilia-cher) [reply recipes_source/recipes/profiler.py:80]: will clarify

File: recipes_source/recipes/profiler.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ilia-cher) [reply recipes_source/recipes/profiler.py:4]: thanks! will need to also change the docs there

File: recipes_source/recipes/profiler.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
+"""
+PyTorch Profiler
+====================================
+This recipe explains how to use PyTorch profiler and measure time and
+memory consumption of model's operators.
+
```
Comment (jlin27) [inline recipes_source/recipes/profiler.py:6]: nit: add the - of "the" model's operators.
Sorry realized missing one more

Comment (ilia-cher) [reply recipes_source/recipes/profiler.py:6]: thanks)

File: recipes_source/recipes/profiler.py (hunk 2 above)
Comment (ilia-cher) [reply recipes_source/recipes/profiler.py:203]: adding a link to TensorBoard viz

**Comment by ilia-cher** [comment]
addressed comments

**Pull request merged**